command/agent/util.go

```diff
@@ -9,3 +11,44 @@ import (
 func randomStagger(intv time.Duration) time.Duration {
 	return time.Duration(uint64(rand.Int63()) % uint64(intv))
 }
+
+// IpOfDevice returns a routable ip addr of a device
+func ipOfDevice(name string) (net.IP, error) {
+	intf, err := net.InterfaceByName(name)
+	if err != nil {
+		return nil, err
+	}
+	addrs, err := intf.Addrs()
+	if err != nil {
+		return nil, err
+	}
+	if len(addrs) == 0 {
+		return nil, fmt.Errorf("no ips were detected on the interface: %v", name)
+	}
+	var ipv4Addrs []net.IP
+	var ipv6Addrs []net.IP
+	for _, addr := range addrs {
+		var ip net.IP
+		switch v := (addr).(type) {
+		case *net.IPNet:
+			ip = v.IP
+			if ip.To4() != nil {
```

Comment:
Do you think there is any merit in ignoring the loopback IP here? Binding to a loopback IP does not make sense since no one else can access it. This might help if the user mistakenly puts something like lo0